### PR TITLE
implementation (Languages): #28456 Add Route and create Angular Library

### LINKED ISSUE #28456: Languages: Add Route and create Angular Library
### Parent Issue

#25604

### User Story

Add new Angular Route to host the Languages migration.

### Acceptance Criteria

- Add new route 
- Create new lIbrary. 

### Proposed Objective

User Experience

### Proposed Priority

Priority 2 - Important

### External Links... Slack Conversations, Support Tickets, Figma Designs, etc.

_No response_

### Assumptions & Initiation Needs

_No response_

### Quality Assurance Notes & Workarounds

_No response_

### Sub-Tasks & Estimates

### PR DESCRIPTION
**Pull request by hmoreras**
### Proposed Changes
* Add angular route for new migration of Languages.

### CODE REVIEW

File: core-web/apps/dotcms-ui/src/app/app-routing.module.ts [hunk 1]
```diff
@@ -60,6 +60,17 @@ const PORTLETS_ANGULAR: Route[] = [
...
             )
     },
+    {
+        path: 'languages',
+
+        canActivate: [MenuGuardService],
+        canActivateChild: [MenuGuardService],
+        data: {
+            reuseRoute: false
+        },
```
Comment (fmontes) [inline core-web/apps/dotcms-ui/src/app/app-routing.module.ts:70]: What is this for?

**Review by fmontes — APPROVED** [review]

**Review by zJaaal — APPROVED** [review]

File: core-web/libs/portlets/dot-languages/portlet/src/lib/dot-languages/dot-languages.component.ts [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import { CommonModule } from '@angular/common';
+import { Component } from '@angular/core';
+
+@Component({
+    selector: 'lib-dot-languages',
+    standalone: true,
+    imports: [CommonModule],
+    templateUrl: './dot-languages.component.html',
+    styleUrl: './dot-languages.component.css'
+})
```
Comment (oidacra) [inline core-web/libs/portlets/dot-languages/portlet/src/lib/dot-languages/dot-languages.component.ts:10]: add OnPush

File: core-web/libs/portlets/dot-languages/portlet/project.json [hunk 3]
```diff
@@ -0,0 +1,17 @@
+{
+    "name": "dot-languages",
+    "$schema": "../../../../node_modules/nx/schemas/project-schema.json",
+    "sourceRoot": "libs/portlets/dot-languages/portlet/src",
+    "prefix": "lib",
+    "tags": [],
```
Comment (oidacra) [inline core-web/libs/portlets/dot-languages/portlet/project.json:6]: Add the tags to the lib.
```
   "tags": ["type:feature", "scope:dotcms-ui", "portlet:languages"]
```

File: core-web/tsconfig.base.json [hunk 4]
```diff
@@ -55,7 +58,8 @@
             "@portlets/*": ["apps/dotcms-ui/src/app/portlets/*"],
             "@services/*": ["apps/dotcms-ui/src/app/api/services/*"],
             "@shared/*": ["apps/dotcms-ui/src/app/shared/*"],
-            "@tests/*": ["apps/dotcms-ui/src/app/test/*"]
+            "@tests/*": ["apps/dotcms-ui/src/app/test/*"],
+            "dot-languages": ["libs/portlets/dot-languages/portlet/src/index.ts"]
```
Comment (oidacra) [inline core-web/tsconfig.base.json:62]: Why exist 2 alias to the same index
`"@dotcms/portlets/dot-languages/portlet": [ "libs/portlets/dot-languages/portlet/src/index.ts" ],`
`"dot-languages": ["libs/portlets/dot-languages/portlet/src/index.ts"]`

File: core-web/apps/dotcms-ui/src/app/app-routing.module.ts (hunk 1 above)
Comment (hmoreras) [reply core-web/apps/dotcms-ui/src/app/app-routing.module.ts:70]: we implement a custom `RouteReuseStrategy` this indicate that if the params change in the same route, do not re-use the route, instead load all again.

File: core-web/libs/portlets/dot-languages/portlet/src/lib/dot-languages/dot-languages.component.spec.ts
Comment (rjvelazco) [inline core-web/libs/portlets/dot-languages/portlet/src/lib/dot-languages/dot-languages.component.spec.ts:1]: Should we use `spectator`?

File: core-web/libs/portlets/dot-languages/portlet/tsconfig.json [hunk 5]
```diff
@@ -0,0 +1,29 @@
+{
+    "compilerOptions": {
+        "target": "es2022",
+        "useDefineForClassFields": false,
+        "forceConsistentCasingInFileNames": true,
+        "strict": true,
+        "noImplicitOverride": true,
+        "noPropertyAccessFromIndexSignature": true,
+        "noImplicitReturns": true,
+        "noFallthroughCasesInSwitch": true
```
Comment (rjvelazco) [inline core-web/libs/portlets/dot-languages/portlet/tsconfig.json:10]: Doesn't this overwrite the `compilerOptions` of the `tsconfig.base.json`?

File: core-web/libs/portlets/dot-languages/portlet/tsconfig.json [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+            "path": "./tsconfig.spec.json"
+        }
+    ],
+    "extends": "../../../../tsconfig.base.json",
+    "angularCompilerOptions": {
+        "enableI18nLegacyMessageIdFormat": false,
+        "strictInjectionParameters": true,
+        "strictInputAccessModifiers": true,
+        "strictTemplates": true
+    }
```
Comment (rjvelazco) [inline core-web/libs/portlets/dot-languages/portlet/tsconfig.json:28]: Doesn't this overwrite the `angularCompilerOptions` of the `tsconfig.base.json`?

File: core-web/tsconfig.base.json (hunk 4 above)
Comment (hmoreras) [reply core-web/tsconfig.base.json:62]: fixed

File: core-web/libs/portlets/dot-languages/portlet/src/lib/dot-languages/dot-languages.component.spec.ts
Comment (hmoreras) [reply core-web/libs/portlets/dot-languages/portlet/src/lib/dot-languages/dot-languages.component.spec.ts:1]: sure, this is a placeholder for now.

File: core-web/libs/portlets/dot-languages/portlet/tsconfig.json (hunk 5 above)
Comment (hmoreras) [reply core-web/libs/portlets/dot-languages/portlet/tsconfig.json:10]: not really, the only `target` is defined in the `tsconfig.base.json`

File: core-web/libs/portlets/dot-languages/portlet/tsconfig.json (hunk 6 above)
Comment (hmoreras) [reply core-web/libs/portlets/dot-languages/portlet/tsconfig.json:28]: ther is no `angularCompilerOptions` in the base

**Review by oidacra — APPROVED** [review]

**Pull request merged**